### ISSUE TITLE
After key expiration, reauthentication removes all tags from the machine

### ISSUE THREAD

**Issue by robmv** [tailscale/tailscale#10776]
### What is the issue?

This is the first time since creating the tailnet that devices' keys are starting to expire. The devices were re-authenticated, but the tags assigned to those devices were lost. I had to assign them again.

### Steps to reproduce

1. Let some key expire on a machine assigned to a tag.
2. Go the expired client and re-authenticate.
3. Tags are gone on the device.

### OS

Linux, Other

### OS version

Fedora release 39 (Thirty Nine)

### Tailscale version

1.56.1-1

### Other software

This initially happened on my devices since I was the person that setup the tailnet. It happened initially on my Linux laptop, and later on my Windows test device. And confirmed on the next person that keys expired on Windows and Linux too.

Looks like re-authentication only preserve the machine name. If every time key expiration means having to reconfigure everything of the machine, then more people will opt to disable key expiration that isn't a very good outcome, specially if the network is large.

**Comment by DentonGentry** [comment]
Were reauthentication done using an [authkey with tags applied](https://tailscale.com/kb/1068/acl-tags#using-an-acl-tag-with-an-auth-key), the newly reauthorized node would have tags.

Were reauthentication done using `tailscale up --tags=tag:example`, the newly reauthorized node would have tags.

It sounds like reauthentication was done by logging in as a User. The new node is owned by the User who created it.

**Comment by robmv** [comment]
This is still happening, My machine got completely unconfigured, tags and accepted routes lost after re-authentication. At least this time the custom name wasn't lost like last time.

For non technical users this makes the entire re-authentication process a hassle. The only workaround is disabling key expiration entirely.

I only did a `tailscale up` (I don´t have tailscale up all the time) it asked me for re-authentication. Followed the URL and approved everything. The machine should have stayed with the same configuration as before. I am afraid the next round of non technical users will have a lot of problems when their authentication is forced next time.

**Comment by matthaywardwebdesign** [comment]
Just flagging to say that we are also seeing this for Mac users. When their keys expire and they re-login all of their tags gets reset. 

is there anyway to avoid this?

**Comment by teleclimber** [comment]
I've experienced this too, and I agree that it's frustrating. According to this issue it works as intended though.

https://github.com/tailscale/tailscale/issues/6997

However I think Tailscale should reconsider the intended behavior.

**Comment by jamesgoodhouse** [comment]
I can confirm this is still an issue, as my phone's key expired, and after re-authing through the mobile app, the tags were no longer attached to the device.

**Comment by smartiq** [comment]
Same thing happened to me today when I had to reauth my iOS phone running 1.90.4. I re-authenticated but couldn't access the network until I logged in and re-tagged my phone. Seems like a bug? In its current form it is a great way to DOS yourself.

**Comment by tjhart85** [comment]
At the very least, it'd be nice if it needed to be re-approved afterwards rather than starting from a blank slate or if the system would send an email that a device was added without ACL's and currently doesn't have access to anything.

I find myself completely disabling the expiration instead and to be honest, I can't imagine that _that_ would be the preferred behavior over simply re-signing in!

It's really weird that software that advertises itself as being so easy to work with makes this one thing so incredibly difficult in comparison to everything else and it's not considered a bug, but a feature, somehow!

**Comment by kbowser21** [comment]
Agree this is a bit of a pain. We use tags to filter allow traffic to different endpoints and when a user has to reauthenticate, they lose all connectivity to the internal network until an admin can tag the user's device again.

**Comment by mberymon** [comment]
I think if this is the intended behavior it should say expired when status is queried rather than logged out which motivates users to log back in to fix it rather than use up or whatever other methods would preserve configs. It's definitely bad UX design.

**Comment by joshmitch2000** [comment]
I agree this is frustrating especially on macos and ios when the key expires. Like others, I've resorted to disabling key expiry on these devices to prevent losing my tags but this seems far from ideal

**Comment by bakugo** [comment]
Just ran into this as well. iOS client forced me to reauthenticate, I did, then proceeded to spend half an hour banging my head against a wall wondering why I couldn't connect to any other node before realizing the device lost its tags. Please fix.
